metrics-caffeine/src/main/java/com/codahale/metrics/caffeine/MetricsStatsCounter.java

```diff
@@ -0,0 +1,110 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.codahale.metrics.caffeine;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.TimeUnit;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.github.benmanes.caffeine.cache.stats.CacheStats;
+import com.github.benmanes.caffeine.cache.stats.StatsCounter;
+
+/**
+ * A {@link StatsCounter} instrumented with Dropwizard Metrics.
+ *
+ * @author ben.manes@gmail.com (Ben Manes)
+ */
+public final class MetricsStatsCounter implements StatsCounter {
+  private final Counter hitCount;
+  private final Counter missCount;
+  private final Counter loadSuccessCount;
+  private final Counter loadFailureCount;
+  private final Timer totalLoadTime;
+  private final Counter evictionCount;
+  private final Counter evictionWeight;
+
+  /**
+   * Constructs an instance for use by a single cache.
+   *
+   * @param registry the registry of metric instances
+   * @param metricsPrefix the prefix name for the metrics
+   */
+  public MetricsStatsCounter(MetricRegistry registry, String metricsPrefix) {
+    requireNonNull(metricsPrefix);
+    hitCount = registry.counter(metricsPrefix + ".hits");
+    missCount = registry.counter(metricsPrefix + ".misses");
+    totalLoadTime = registry.timer(metricsPrefix + ".loads");
+    loadSuccessCount = registry.counter(metricsPrefix + ".loads-success");
+    loadFailureCount = registry.counter(metricsPrefix + ".loads-failure");
+    evictionCount = registry.counter(metricsPrefix + ".evictions");
+    evictionWeight = registry.counter(metricsPrefix + ".evictions-weight");
+  }
+
+  @Override
+  public void recordHits(int count) {
+    hitCount.inc(count);
+  }
+
+  @Override
+  public void recordMisses(int count) {
+    missCount.inc(count);
+  }
+
+  @Override
+  public void recordLoadSuccess(long loadTime) {
+    loadSuccessCount.inc();
+    totalLoadTime.update(loadTime, TimeUnit.NANOSECONDS);
+  }
+
+  @Override
+  public void recordLoadFailure(long loadTime) {
+    loadFailureCount.inc();
+    totalLoadTime.update(loadTime, TimeUnit.NANOSECONDS);
+  }
+
+  @Override
+  @SuppressWarnings("deprecation")
+  public void recordEviction() {
+    // This method is scheduled for removal in version 3.0 in favor of recordEviction(weight)
+    recordEviction(1);
+  }
+
+  @Override
+  public void recordEviction(int weight) {
+    evictionCount.inc();
+    evictionWeight.inc(weight);
+  }
```

Comment:
fyi, the example might be built on a older version of caffeine (2.7.0). 

In 2.8.0 (via https://github.com/ben-manes/caffeine/issues/304) we added the `RemovalCause` as a parameter. You might want to update to that delegate, though it has no direct bearing on your usage so it will be identical behavior.

```java
@Deprecated
default void recordEviction(@NonNegative int weight) {
  recordEviction();
}

/**
 * Records the eviction of an entry from the cache. This should only been called when an entry is
 * evicted due to the cache's eviction strategy, and not as a result of manual
 * {@link Cache#invalidate invalidations}.
 *
 * @param weight the weight of the evicted entry
 * @param cause the reason for which the entry was removed
 */
default void recordEviction(@NonNegative int weight, RemovalCause cause) {
  // This method will be abstract in version 3.0.0
  recordEviction(weight);
}
```

Reply:
Yeah, I saw that. I think assigning causes would be a use case for tagged metrics, but that's not supported in the 4.1.x branch this will be merged into. I don't know the dropwizard release management flow, but perhaps after this is merged, it could also be applied to the 5.x branch, and then there could be a followup PR against 5.x to tag with RemovalCause..

Reply:
@john-karp The 5.x branch of Dropwizard Metrics is currently on ice and is not actively developed.

What do you think about having a counter for the total number of evictions (`metricsPrefix + ".evictions"` and `metricsPrefix + ".evictions-weight"`) and one counter for each removal cause (`metricsPrefix + ".evictions." + cause` and `metricsPrefix + ".evictions-weight." + cause`, for example `mycache.evictions.size` and `mycache.evictions-weight.explicit`)?

Reply:
OK, added metrics breaking down evictions by cause.